### DISCUSSION TITLE
Logging / Debugging?

### DISCUSSION THREAD

**Discussion by jonstelly** [git-ecosystem/git-credential-manager#1489] [Q&A]
There doesn't seem to be any diagnostic/trace logging in GCM, am I maybe missing something?  I'm having an issue and it's REALLY difficult trying to track down what's going on, where if there was some sort of debug/trace flag I could set, this would have been much easier...

I've got a WSL instance of Arch Linux, standalone install of git and gcm inside the WSL instance... I've had this set up and working in the past but had to rebuild the machine and must have missed something.

When GCM should prompt me to login (git clone, git pull, etc...), it is opening Windows Explorer to my Documents folder.  From looking through the GCM code I see some references to xdg-open, I verified that if I 'xdg-open browser' it opens Windows Explorer to my Documents folder so this seems like the right track.  I still don't know if this is exactly the problem, I think I'm going to need to build a debug version of GCM to track down exactly what configuration I missed when rebuilding my computer.  But this would be much easier if I could enable logging in GCM to see what it's doing, it's a bit too much of a black-box IMHO.

**Comment by hickford** [comment]
Have you tried setting environment variable [GCM_TRACE](https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/environment.md#gcm_trace)=1?

> Enables trace logging of all activities. Configuring Git and GCM to trace to the same location is often desirable, and GCM is compatible and cooperative with [GIT_TRACE](https://git-scm.com/docs/git#Documentation/git.txt-codeGITTRACEcode).
